### ISSUE TITLE
Treat typedef/type-alias with TSTypes highlight

### ISSUE THREAD

**Issue by GGCristo** [nvim-treesitter/nvim-treesitter#633]
With no configuration its look like this.
![vim2](https://user-images.githubusercontent.com/58046649/98056187-3075e780-1e37-11eb-8050-97be304bfc67.png)
When I change the TSVariable its turns like this.
![vim1](https://user-images.githubusercontent.com/58046649/98056184-2eac2400-1e37-11eb-9f8a-d1d1a01c7055.png)
But ```sf::Vector2f``` is a typedef, not a variable.

**Comment by vigoux** [comment]
Hi, thanks for reporting, could you try running `:TSHighlightCapturesUnderCursor` whith the cursor on `sf::Vector2f` to see what highlights are used there ?
Thanks !

**Comment by theHamsta** [comment]
It's probably the constructor highlight. I map constructor to TSType because I had the same issue.

**Comment by GGCristo** [comment]
```
@function -> TSFunction
@constructor -> TSConstructo
@type -> TSType
@variable -> TSVariable
```

**Comment by theHamsta** [comment]
You basically get a mixture of the four highlights. C highlights should be included before C++ highlights. But it's now the other way around, whenever I try it?@vigoux did something change with the `inherits` syntax? `@variable` is the very first query in C

**Comment by vigoux** [comment]
I don't think we should expect the output of those to be ordered in the order they appear tbh.
Maybe the issue is caused by overlapping matches.

**Comment by GGCristo** [comment]
![vim](https://user-images.githubusercontent.com/58046649/98117191-44eeca00-1ea1-11eb-9874-7152d7bb67f9.png)
I am just a user but I kind of understand what you are saying here, the priority is messed up right? In more complex syntax than the one in the photo it tries to get the TSVariable highlight, but when this one is not set it use the TSConstructor.
Thank you bot for the quick response, I hope this is trivial to solve.

**Comment by vigoux** [comment]
Hmm, coming back to this, I think we'd need a small repro file to help out a bit, do you have one ?
This way we can work our way into fixing the issue.

**Comment by GGCristo** [comment]
With SFML is as simple as doing:
```cpp
#include <vector>
#include <SFML/System/Vector2.hpp>
int main()
{
  std::vector<sf::Vector2u> vvector;
  vvector.push_back(sf::Vector2u());
  return 0;
}
```
I am still trying to reproduce this without the need of SFML's API, but maybe it takes a while because I am busy with University.

Trying to reproduce the bug I also notice that when you overload an operator if you don't leave a space between the operator symbol and the parenthesis and one of the parameter is not a primitive variable the highlight doesn't work:

![todobien](https://user-images.githubusercontent.com/58046649/98406043-2802f400-2065-11eb-94df-39b0cf227c6d.png)
![mal](https://user-images.githubusercontent.com/58046649/98406048-2a654e00-2065-11eb-91f5-93db461411bb.png)

**Comment by theHamsta** [comment]
@vigoux found the bug: https://github.com/neovim/neovim/blob/d17e508796be60eefe4a597df62de1fd9e7e1725/runtime/lua/vim/treesitter/query.lua#L76-L79

you're putting the base files after the actual language file. This inverted the order of the queries and place `@variable` after all c++ queries.

**Comment by theHamsta** [comment]
Fixed via  #13253
